### PR TITLE
Stability fixes

### PR DESCRIPTION
**Pull request by runcom**
these are patches taken from the kube-1.6.x branch - these patches are the result of many testing to make CRI-O works with kubeadm.
I also opened this to make sure kube tests are ok.

@sameo @mrunalp PTAL

### CODE REVIEW

File: oci/oci.go [hunk 1]
```diff
@@ -550,13 +550,10 @@ func (r *Runtime) UpdateStatus(c *Container) error {
...
 	out, err := exec.Command(r.Path(c), "state", c.name).CombinedOutput()
 	if err != nil {
-		if err := unix.Kill(c.state.Pid, 0); err == syscall.ESRCH {
-			c.state.Status = ContainerStateStopped
-			c.state.Finished = time.Now()
-			c.state.ExitCode = 255
-			return nil
-		}
-		return fmt.Errorf("error getting container state for %s: %s: %q", c.name, err, out)
+		c.state.Status = ContainerStateStopped
```
Comment (mrunalp) [inline oci/oci.go]: A comment here would be useful for posterity :)

File: server/sandbox_run.go [hunk 2]
```diff
@@ -84,16 +89,38 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 
 	id, name, err := s.generatePodIDandName(kubeName, namespace, attempt)
 	if err != nil {
-		return nil, err
+		if strings.Contains(err.Error(), "already reserved for pod") {
```
Comment (mrunalp) [inline server/sandbox_run.go]: We could probably introduce a custom error here. (could be handled in follow-on).

**Comment by mrunalp** [comment]
LGTM

File: server/sandbox_run.go (hunk 2 above)
Comment (sameo) [inline server/sandbox_run.go]: Yes, it would be great to track an error variable rather than a open coded string.

**Comment by sameo** [comment]
Tested on a bare metal k8s deployment, works fine.

**Comment by sameo** [comment]
LGTM

File: server/sandbox_run.go (hunk 2 above)
Comment (runcom) [inline server/sandbox_run.go]: we can actually remove this and just use the `regexp` :)

Comment (runcom) [inline server/sandbox_run.go]: if `matches` isn't 2 just return the error (as it didn't match that error)

**Pull request merged**